imod/mf6/simulation.py

```diff
@@ -941,12 +941,22 @@ def split(self, submodel_labels: xr.DataArray) -> Modflow6Simulation:
                 new_simulation[solution_name].add_model_to_solution(new_model_name)
 
         exchanges = []
-        for model_name, model in original_models.items():
-            if isinstance(model, GroundwaterFlowModel):
-                exchanges += exchange_creator.create_gwfgwf_exchanges(
-                    model_name, model.domain.layer
+        flow_models = self.get_models_of_type("gwf6")
```

Comment:
This is a check that you want to do as early as possible. You can move it to somewhere at the top of this method.

Reply:
done